pkg/types/vsphere/validation/platform.go

```diff
@@ -49,7 +51,7 @@ func ValidatePlatform(p *vsphere.Platform, fldPath *field.Path) field.ErrorList
 }
 
 // ValidateForProvisioning checks that the specified platform is valid.
-func ValidateForProvisioning(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
+func ValidateForProvisioning(p *vsphere.Platform, network *types.Networking, fldPath *field.Path) field.ErrorList {
```

Comment:
`network` is unused now.
```suggestion
func ValidateForProvisioning(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
```